pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/QueryRunner.java

```diff
@@ -94,12 +98,19 @@ public void processQuery(DistributedStagePlan distributedStagePlan, ExecutorServ
           ServerRequestUtils.constructServerQueryRequest(distributedStagePlan, requestMetadataMap);
 
       // send the data table via mailbox in one-off fashion (e.g. no block-level split, one data table/partition key)
-      DataTable dataTable = _serverExecutor.processQuery(serverQueryRequest, executorService, null);
+      BaseDataBlock dataBlock;
+      try {
+        DataTable dataTable = _serverExecutor.processQuery(serverQueryRequest, executorService, null);
```

Comment:
I wanted to comment on the else block but for some reason github is not letting me ....

The else part does not seem correct to me. I agree that if this stage is executing on leaf servers, then we can the ServerQueryExecutor, operators, DataTable (from today's code) and build a transferable block to send to the next stage.

However, if this is an intermediary stage, then we still expect stages downstream and should still be sending TransferableBlock to either another intermediary stage or the final broker reduce / merge stage ?